src/openzeppelin/utils/presets/UniversalDeployer.cairo

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.4.0b (utils/presets/UniversalDeployer.cairo)
+
+%lang starknet
+
+from starkware.starknet.common.syscalls import get_caller_address, deploy
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.hash import hash2
+from starkware.cairo.common.bool import FALSE, TRUE
+
+@event
+func ContractDeployed(
+    address: felt,
+    deployer: felt,
+    unique: felt,
+    classHash: felt,
+    calldata_len: felt,
+    calldata: felt*,
+    salt: felt
+) {
+}
+
+@external
+func deployContract{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    range_check_ptr
+}(
+    classHash: felt,
+    salt: felt,
+    unique: felt,
+    calldata_len: felt,
+    calldata: felt*
+) -> (address: felt) {
+    alloc_locals;
+    let (deployer) = get_caller_address();
+
+    tempvar prefix;
+    tempvar from_zero;
+    if (unique == TRUE) {
+        prefix = deployer;
+        from_zero = FALSE;
+    } else {
+        prefix = 'UniversalDeployerContract';
```

Comment:
Do we really need to keep `UniversalDeployerContract` prefix when `deploy_from_zero=TRUE`. It made sense before the last commits but I feel that now it just adds an unnecessary step to the computation of the address.

When `deploy_from_zero=TRUE` I would just use the user provided salt as is.

Reply:
Agree to this. Not sure why we need it but looking at the conversation [here](https://community.starknet.io/t/universal-deployer-contract-proposal/1864/18), @bbrandtom suggested to have a prefix.

Reply:
yeah I noticed the same but I also thought it wasn't a big issue, otherwise it's a variable revocation mess. will try to update.

Reply:
😭😭😭. any ideas on how to improve it?
https://github.com/OpenZeppelin/cairo-contracts/blob/8632ef03f277ae02259f5e7292dace780cd623c4/src/openzeppelin/utils/presets/UniversalDeployer.cairo#L35-L59